docker/Dockerfile.new

```diff
@@ -0,0 +1,100 @@
+ARG SGX_MODE=HW
+ARG NETWORK_ID=AB
+
+FROM ubuntu:18.04 AS RUNTIME_BASE
+LABEL maintainer="blockchain@crypto.com"
+
+ARG SGX_ENCLAVE=https://download.01.org/intel-sgx/sgx-linux/2.7.1/distro/ubuntu18.04-server/libsgx-enclave-common_2.7.101.3-bionic1_amd64.deb
+RUN set -e; \
+    apt-get update; \
+    apt-get install -y libzmq3-dev libssl1.1 libprotobuf10; \
+    apt-get install -y curl; \
+    mkdir -p /opt/intel; \
+    curl ${SGX_ENCLAVE} -o /tmp/sgx.deb; \
+    dpkg -i /tmp/sgx.deb; \
+    rm /tmp/sgx.deb; \
+    apt-get purge --auto-remove -y curl; \
+    rm -rf /var/lib/apt/lists/*
+
+COPY --from=tendermint/tendermint:v0.32.9 /usr/bin/tendermint /usr/bin/tendermint
+
+FROM baiduxlab/sgx-rust:1804-1.1.0 AS BUILDER_BASE
+LABEL maintainer="blockchain@crypto.com"
+
+ARG SGX_MODE
+ARG NETWORK_ID
+
+ENV SGX_SDK=/opt/sgxsdk
+ENV PATH=/root/.cargo/bin:/root/.local/bin:$PATH
+ENV RUST_BACKTRACE=1
+ENV RUSTFLAGS "-Ctarget-feature=+aes,+sse2,+sse4.1,+ssse3"
+ENV SGX_MODE=$SGX_MODE
+ENV NETWORK_ID=$NETWORK_ID
+
+RUN set -e; \
+    apt-get update -y; \
+    apt-get install -y \
+      cmake \
+      libgflags-dev \
+      libzmq3-dev \
+      pkg-config \
+      clang; \
+    rm -rf /var/lib/apt/lists/*
+
+COPY --from=tendermint/tendermint:v0.32.9 /usr/bin/tendermint /usr/bin/tendermint
+
+FROM BUILDER_BASE AS TEST
+LABEL maintainer="blockchain@crypto.com"
```

Comment:
How will this TEST image be used?

Reply:
It's used as a base image for integration test in drone.

Reply:
Ok, but when we build the test image we have to modify this Dockerfile?

Reply:
> Ok, but when we build the test image we have to modify this Dockerfile?

No need to modify, just `docker build --target TEST -t ... .`